File: sungwoopark95/0x0c/boj_15650.cpp
#include <iostream>
using namespace std;

int N, M;
int arr[10];
bool isused[10];

void func(int cur) {
    if (cur == M) {
        for (int i=0;i<M;i++) {
            cout << arr[i] << " ";
        }
        cout << "\n";
        return;
    }
    
    for (int i=1;i<=N;i++) {
        if(!isused[i-1]) {
            if (cur >= 1 && arr[cur-1] > i) { continue; }
            arr[cur] = i;
            isused[i-1] = 1;
            func(cur+1);
            isused[i-1] = 0;
        }
    }
}

int main() {
    cin.tie(0);
    ios_base::sync_with_stdio(0);
    
    cin >> N >> M;
    func(0);
    
    return 0;
}

File: sungwoopark95/0x0c/boj_15650.py
global arr, isused, N, M

def func(cur):
    if cur == M:
        for i in range(M):
            print(arr[i], end=' ')
        print('')
        return
    
    for i in range(1, N+1):
        if not isused[i-1]:
            if (cur >= 1) and (arr[cur-1] > i): 
                continue
            arr[cur] = i
            isused[i-1] = 1
            func(cur+1)
            isused[i-1] = 0


arr = [0 for i in range(10)]
isused = [0 for i in range(10)]

N, M = tuple(input().split(' '))
N, M = int(N), int(M)
func(0)